api/src/main/java/com/basistech/rosette/RosetteFactoryConfigurationProperty.java

```diff
@@ -0,0 +1,40 @@
+/*
+* Copyright 2017 Basis Technology Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.basistech.rosette;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD})
+/**
+ * This annotation marks factory configuration options that should be included in configuration
+ * files in an on-premise deployment.
+ */
+public @interface RosetteFactoryConfigurationProperty {
```

Comment:
Why?  You haven't actually provided any logical reason to this. To save two lines of code in an annotator we may or may not ever create?  You seriously would rather rely on the *naming scheme of files*!!! Why are you insisting on throwing out a perfectly good solution to this problem, which I already have working?